File: Commons/AOP Tests/src/test/java/com/greentree/commons/tests/aop/TestAOP.java
package com.greentree.commons.tests.aop;

import org.junit.jupiter.api.Nested;

import static org.junit.jupiter.api.Assertions.*;

@AutowiredConfig(TestConfig.class)
public class TestAOP {

    @Nested
    class NestedAOP {

        @AutowiredTest
        void shapeAreaMoreZero(Shape shape) {
            assertTrue(shape.getArea() > 0f);
        }

    }

    @AutowiredTest
    void notNull(Person person) {
        assertNotNull(person.getName());
    }

    @AutowiredTest
    void testTagAnton(@AutowiredArgument(tags = {"Anton"}) Person person) {
        assertEquals(person.getName(), "Anton");
    }

    @AutowiredTest
    void doubleArgumentNotNull(Person person, Shape shape) {
        assertNotNull(person);
        assertNotNull(person.getName());
        assertNotNull(shape);
    }

    @AutowiredTest
    void shapeAreaMoreZero(Shape shape) {
        assertTrue(shape.getArea() > 0f);
    }

}

File: Commons/Context/Tests/src/test/java/aop/TestAOP.kt
package aop

import com.greentree.engine.rex.fuse.tests.AutowiredTest
import com.greentree.engine.rex.fuse.tests.ContextTest
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Nested

@ContextTest(TestConfig::class)
class TestAOP {

	@Nested
	internal inner class NestedAOP {

		@AutowiredTest
		fun shapeAreaMoreZero(shape: Shape) {
			Assertions.assertTrue(shape.area > 0f)
		}
	}

	@AutowiredTest
	fun notNull(person: Person) {
		Assertions.assertNotNull(person.name)
	}

	@AutowiredTest
	fun doubleArgumentNotNull(person: Person?, shape: Shape?) {
		Assertions.assertNotNull(person)
		Assertions.assertNotNull(person!!.name)
		Assertions.assertNotNull(shape)
	}

	@AutowiredTest
	fun shapeAreaMoreZero(shape: Shape) {
		Assertions.assertTrue(shape.area > 0f)
	}
}
